boltstub_rs/src/values/bolt_struct/unsupported_type.rs

```diff
@@ -66,9 +67,12 @@ impl JoltUnsupportedType {
     }
 
     pub(crate) fn into_struct(self) -> PackStreamStruct {
-        let mut extra_map = IndexMap::with_capacity(1);
+        let mut extra_map;
         if let Some(message) = self.message {
+            extra_map = IndexMap::with_capacity(1);
             extra_map.insert(String::from("message"), PackStreamValue::String(message));
+        } else {
+            extra_map = IndexMap::with_capacity(0);
         }
```

Comment:
```suggestion
        let extra_map = match self.message {
            Some(message) => {
                IndexMap::from([(String::from("message"), PackStreamValue::String(message))])
            }
            None => IndexMap::new(),
        };
```
slightly preferable, I find. a) doesn't leave a variable mut for longer than needed b) (in my view) ever so slightly more readable.